### ISSUE TITLE
Create Command Packages with dotnet-pack

### ISSUE THREAD

**Issue by piotrpMSFT** [dotnet/cli#365]
Now that we can load commands, like dotnet-test-xunit, from a NuGet package we should enable a path for creating such NuGet packages. Command packages should contain:
* The command implementation [.dll]
* Dependency paths [.deps]
* Hosts [one per target OS]
  * Ideally we'd have a better story for this. I'm working on finding one but want to unblock the scenario first.

The changes to dotnet-pack should be fairly straightforward but the convention for enabling this behavior is worthy of discussion. I propose we key off of emitEntryPoint:true. If this flag is set then we know that the binary produced is an executable and the only known scenario for NuGet packaging an executable is for it to be treated as a command.

Since commands must follow the naming convention of dotnet-{command} I think we can use the name of the executable as a second check for enabling this packaging scenario. If the file name is not prepended with `dotnet-` then it is not a command and should fall back to existing behavior.

So, in summary:

````
if (project.CompilerOptions.EmitEntryPoint && project.Name.StartsWith("dotnet-"))
{
    package.Add(hosts);
    package.Add(depsFile);
}
````

/cc @davidfowl @anurse @victorhurdugaci @bricelam @livarcocc

**Comment by brthor** [comment]
Is this intended to be only used for CLI command scenarios? The "dotnet-" check seems to imply that. Why not do this for executables in general?

**Comment by piotrpMSFT** [comment]
The scenario, at least for the time being, is to make dotnet ecosystem more easily expandable. The entry point to these applications is the dotnet driver. Using these packages is done via `dotnet {command}`. Since that is a mandatory calling convention for the time being, making that explicit in the executable is a nice convention.

**Comment by blackdwarf** [comment]
@brthor what is the scenario you were thinking when you mention "executables in general"?

**Comment by johanste** [comment]
@blackdwarf, one scenario would be having another "family" of dispatcher/extensibility commands - just as dotnet triggers off dotnet-<executable>, another family might have another named dispatcher with exactly the same (or very similar) extensibility mechanism as what dotnet uses. 

In fact, I'm doing something very similar right now :)

**Comment by blackdwarf** [comment]
Closing out.
